tests/Integration/Schema/Directives/BuilderDirectiveTest.php

```diff
@@ -155,6 +156,47 @@ public function testCallsCustomBuilderMethodOnFieldWithoutValue(): void
         ')->assertJsonCount(2, 'data.users');
     }
 
+    public function testCallsCustomBuilderMethodOnFieldWithSpecificModel(): void
+    {
+        $this->schema = /** @lang GraphQL */ <<<GRAPHQL
+        type Query {
+            users: [User!]! @all
+        }
+
+        type User {
+            id: ID
+            tasks: [Task!]! @hasMany(type: SIMPLE) @builder(method: "{$this->qualifyTestResolver('specificModel')}")
+        }
+
+        type Task {
+            id: ID
+        }
+        GRAPHQL;
+
+        $users = factory(User::class, 2)->create();
+
+        foreach ($users as $user) {
+            $tasks = factory(Task::class, 3)->make();
+            $tasks[0]->name = $user->name;
+            $user->tasks()->saveMany($tasks);
+        }
```

Comment:
```suggestion
        $users = factory(User::class, 2)->create();

        foreach ($users as $user) {
            assert($user instanceof User);

            $taskWithSameName = factory(Task::class)->make();
            assert($taskWithSameName instanceof Task);
            $taskWithSameName->name = $user->name;
            $taskWithSameName->user()->associate($user);
            $taskWithSameName->save();

            $taskWithOtherName = factory(Task::class)->make();
            assert($taskWithOtherName instanceof Task);
            $taskWithOtherName->name = "Different from {$user->name}";
            $taskWithOtherName->user()->associate($user);
            $taskWithOtherName->save();
        }
```

Could be flaky otherwise.